build.gradle

```diff
@@ -57,29 +57,43 @@ if (project.hasProperty("teamcity")) {
 sourceCompatibility = 1.8
 targetCompatibility = 1.8
 
+/**
+ * When adding new dependencies, we must be mindful of scope, eg api or implementation.
+ * The former will be exposed transitively to our consumers compile classpath.
+ *
+ * If any types from a 3rd-party library are used by us as superclasses, method parameters, or method return types,
+ * that library should be added under api scope.
+ * On the other hand, if our usage of a 3rd-party library is limited to private members or method bodies,
+ * we can mark that library as implementation scope.
```

Comment:
it's in the gradle docs, but let's also add that we should prefer using `implementation` over `api` whenever possible

Reply:
cool i'll add that